File: android/src/main/kotlin/com/diamond/printer/diamond_printer/protocol/PrinterCommandGenerator.kt
package com.diamond.printer.diamond_printer.protocol

import android.graphics.Bitmap

/**
 * Base interface for printer command generators
 */
interface PrinterCommandGenerator {
    fun generateTextCommand(text: String, alignment: String? = null): ByteArray
    fun generateImageCommand(bitmap: Bitmap, maxWidth: Int = 576, alignment: String? = null): ByteArray
    fun generateCutCommand(): ByteArray
    fun generateFeedCommand(lines: Int = 3): ByteArray
}

File: ios/Classes/Protocol/PrinterCommandGenerator.swift
import Foundation
import UIKit

/// Base protocol for printer command generators
protocol PrinterCommandGenerator {
    func generateTextCommand(_ text: String, alignment: String?) -> Data
    func generateImageCommand(_ image: UIImage, maxWidth: Int, alignment: String?) -> Data
    func generateCutCommand() -> Data
    func generateFeedCommand(lines: Int) -> Data
}

